src/System.Net.Primitives/src/System/Net/IPEndPoint.cs

```diff
@@ -119,6 +119,84 @@ public int Port
             }
         }
 
+        public static bool TryParse(string endPointString, out IPEndPoint result)
+        {
+            return TryParse(endPointString.AsSpan(), out result);
+        }
+
+        public static bool TryParse(ReadOnlySpan<char> endPointSpan, out IPEndPoint result)
+        {
+            result = null;
+            int sliceLength = endPointSpan.Length;  // If there's no port then send the entire string to the address parser
+            int lastColonPos = endPointSpan.LastIndexOf(':');
+
+            // Look to see if this is an IPv6 address with a port.
+            if (lastColonPos > 0 && endPointSpan[lastColonPos - 1] == ']')
+            {
+                sliceLength = lastColonPos;
+            }
+            else if(lastColonPos > 0)
```

Comment:
There's no need to check lastColonPos twice... it can instead be:
```C#
if (lastColonPos > 0)
{
    if (endPointSpan[lastColonPos - 1] == ']')
    {
        ...
    }
    else
    {
        ...
    }
}
...
```